Starcounter.Uniform.Tests/Builder/UniDataTableTests.cs

```diff
@@ -0,0 +1,155 @@
+{
+    public class UniDataTableTests
+    {
+        private UniDataTable _sut;
+        private Mock<IFilteredDataProvider<Json>> _dataProviderMock;
+        private List<DataTableColumn> _dataTableColumns;
+        private int _initialPageSize;
+        private int _initialPageIndex;
+        private Func<IReadOnlyCollection<Json>> _returnedRowsFunc;
+        private string _columnPropertyName = "FirstName";
+
+        [SetUp]
+        public void SetUp()
+        {
+            _dataProviderMock = new Mock<IFilteredDataProvider<Json>>();
+            _dataProviderMock.SetupAllProperties();
+            _returnedRowsFunc = () => new List<Json>();
+            _dataProviderMock.Setup(provider => provider.CurrentPageRows).Returns(() => _returnedRowsFunc());
+            _dataTableColumns = new List<DataTableColumn>();
+            _initialPageSize = 20;
+            _initialPageIndex = 0;
+        }
+
+        private void InitSut()
+        {
+            _sut = new UniDataTable().Init(_dataProviderMock.Object, _dataTableColumns, _initialPageSize, _initialPageIndex);
+        }
+
+        [Test]
+        public void AfterInitSpecifiedColumnShouldBePresent()
+        {
+            var columnDisplayName = "First name";
+            _dataTableColumns.Add(new DataTableColumn { DisplayName = columnDisplayName });
+
+            InitSut();
+
+            _sut.Columns.Should().HaveCount(1);
+            _sut.Columns.Should().ContainSingle().Which.DisplayName.Should().Be(columnDisplayName);
+            _sut.Columns.Should().ContainSingle().Which.LoadRows.Should().NotBeNull();
+        }
+
+        [Test]
+        public void AfterInitInitialPageShouldBeLoaded()
+        {
+            var expectedRow = new Json();
+            var expectedRowsCollection = new List<Json>
+            {
+                expectedRow
+            };
+            var initialPageIndex = 3;
+            _initialPageIndex = initialPageIndex;
+            _dataProviderMock
+                .Setup(provider => provider.CurrentPageRows)
+                .Returns(() =>
+                _dataProviderMock.Object.PaginationConfiguration
+                    .CurrentPageIndex == initialPageIndex
+                    ? expectedRowsCollection
+                    : new List<Json>());
+
+            InitSut();
+
+            _sut.Pages.Should().HaveCount(initialPageIndex + 1);
+            _sut.Pages[initialPageIndex].Rows.Should().ContainSingle().Which.Should().BeSameAs(expectedRow);
+        }
+
+        [Test]
+        public void AfterChangingCurrentPageIndexPaginationConfigurationShouldChange()
+        {
+            var newPageIndex = 2;
+
+            InitSut();
+            _sut.Pagination.Handle(new UniDataTable.PaginationViewModel.Input.CurrentPageIndex { Value = newPageIndex });
+
+            _dataProviderMock.Object.PaginationConfiguration.CurrentPageIndex.Should().Be(newPageIndex);
+            _dataProviderMock.VerifyGet(x => x.CurrentPageRows, Times.Exactly(2));
```

Comment:
you only check that rows were retrieved twice. You should check that after changing the page index new rows were retrieved and exposed. This test (in its current form) would pass if, in `Init` I called `CurrentPageRows` twice.

Reply:
I'll change the test to check if correct ammout of pages were retrieved.